### PR TITLE
locale_gen: refactor module to use CmdRunner

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Refactor module to use `CmdRunner`.

##### ISSUE TYPE

- Refactoring Pull Request

##### COMPONENT NAME

locale_gen

### CODE REVIEW

File: plugins/modules/locale_gen.py [hunk 1]
```diff
@@ -174,13 +182,13 @@ def apply_change_ubuntu(targetState, name):
...
-        localeGenExitValue = call(["locale-gen", "--purge"])
-
-    if localeGenExitValue != 0:
-        raise EnvironmentError(localeGenExitValue, "locale.gen failed to execute, it returned " + str(localeGenExitValue))
+        with runner("purge") as ctx:
+            ctx.run()
 
 
 def main():
+    global module
```
Comment (felixfontein) [inline plugins/modules/locale_gen.py:190]: How much work would it be to get rid of that global variable completely?

**Comment by felixfontein** [comment]
(Also this needs a changelog fragment :) )

File: plugins/modules/locale_gen.py (hunk 1 above)
Comment (russoz) [reply plugins/modules/locale_gen.py:190]: going to make this a MH module, but going in small bites ;-) - the global will disappear soon

**Comment by russoz** [comment]
yup, I am waiting for the CI to run, to make sure we're good

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks!

**Pull request merged**